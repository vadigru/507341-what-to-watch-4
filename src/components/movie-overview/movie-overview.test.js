import React from "react";
import renderer from "react-test-renderer";
import MovieOverview from "./movie-overview.jsx";

const movie = {
  title: `Movie Title`,
  posterUrl: `https://url.com/poster.jpg`,
  backgroundUrl: `https://url.com/poster/1.jpg`,
  previewUrl: `https://url.com/preview/video.mp4`,
  genre: `Movie Genre`,
  release: 2020,
  director: `Director Name`,
  starring: [`Actor One`, `Actor Two`, `Actor Three`],
  time: `1h 00m`,
  rating: 10,
  votes: 1000,
  description: `Movie Description`,
  reviews: [
    {
      date: `June 25, 2020`,
      user: `John Doe`,
      comment: `Comment text.`,
      rating: 8.9
    },
  ]
};

it(`Should render MovieOverview component`, () => {
  const tree = renderer
    .create(
        <MovieOverview
          movie={movie}
        />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
