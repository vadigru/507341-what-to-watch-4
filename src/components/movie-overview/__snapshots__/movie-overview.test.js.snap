// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieOverview component 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      10
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Awesome
      </span>
      <span
        className="movie-rating__count"
      >
        1000000
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Some Description
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Famous Director
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring:
        Actor One, Actor Two, Actor Three
         and other
      </strong>
    </p>
  </div>,
]
`;
